sqld/src/database/dump_loader.rs

```diff
@@ -0,0 +1,91 @@
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::PathBuf;
+
+use anyhow::anyhow;
+use sqld_libsql_bindings::wal_hook::WalHook;
+use tokio::sync::{mpsc, oneshot};
+
+use super::libsql::open_db;
+
+type OpMsg = Box<dyn FnOnce(&rusqlite::Connection) + 'static + Send + Sync>;
+
+pub struct DumpLoader {
+    sender: mpsc::Sender<OpMsg>,
+}
+
+impl DumpLoader {
+    pub async fn new(
+        path: PathBuf,
+        hooks: impl WalHook + Clone + Send + 'static,
+    ) -> anyhow::Result<Self> {
+        let (sender, mut receiver) = mpsc::channel::<OpMsg>(1);
+
+        let (ok_snd, ok_rcv) = oneshot::channel::<anyhow::Result<()>>();
+        tokio::task::spawn_blocking(move || {
+            let db = match open_db(path, hooks, false) {
+                Ok(db) => {
+                    let _ = ok_snd.send(Ok(()));
+                    db
+                }
+                Err(e) => {
+                    let _ = ok_snd.send(Err(e));
+                    return;
+                }
+            };
+            while let Some(f) = receiver.blocking_recv() {
+                f(&db);
+            }
```

Comment:
Just out of curiosity, to check if I got things right.
This will be executed only once, right?

Reply:
once for every call to load dump